src/tools/ApiChecker/IdentifyMissingTests.cs

```diff
@@ -0,0 +1,164 @@
+/*
+ * MIT License
+ * 
+ * Copyright (c) Microsoft Corporation.
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using CommandLine;
+
+namespace ApiChecker
+{
+    /// <summary>
+    /// This will identify missing tests from upstream.
+    /// </summary>
+    internal static class IdentifyMissingTests
+    {
+        private static readonly List<(string FileName, string TestName)> _testPairs = new();
+
+        /// <summary>
+        /// Runs the scenario.
+        /// </summary>
+        /// <param name="options">The options argument.</param>
+        public static void Run(IdentifyMissingTestsOptions options)
+        {
+            // get all files that match a pattern
+            var directoryInfo = new DirectoryInfo(options.SpecFileLocations);
+            if (!directoryInfo.Exists)
+            {
+                throw new ArgumentException($"The location ({directoryInfo.FullName}) specified does not exist.");
+            }
+
+            // let's map the test cases from the spec files
+            MapTestsCases(directoryInfo, options);
+
+            // now, let's load the DLL and use some reflection-fu
+            var assembly = Assembly.Load(options.TestDLLName);
```

Comment:
Do we need to load the assembly if you are referencing the project?

Reply:
Fair enough, I've replaced it with `GetAssembly` from the attribute type.